.specify/memory/constitution.md

```diff
@@ -898,6 +959,77 @@ patient data breaches and legal penalties.
 
 ---
 
+### IX. Spec-Driven Iteration (ADDED 2025-12-04)
+
+**MANDATE**: Features requiring >3 days effort MUST be broken into Validation
+Milestones. Each Milestone = 1 Pull Request. Milestones can be parallel `[P]` or
+sequential, enabling flexible team coordination.
+
+**Rules**:
+
+- **Spec First**: Specification PR created first on `spec/{issue-id}-{name}`
+  branch
+- **Milestone = PR**: Each milestone generates exactly one pull request
+- **Parallel Milestones `[P]`**: Can be developed simultaneously by different
+  developers or in any order
+- **Sequential Milestones**: Must complete in order due to dependencies
+- **Branch per Milestone**: Each milestone gets its own branch following naming
+  convention
+- **Verification Gate**: Each milestone PR must pass its verification criteria
+  before merge
+
+**Branch Naming Convention**:
+
+| Branch Type         | Pattern                              | Example                                   |
+| ------------------- | ------------------------------------ | ----------------------------------------- |
+| Spec Branch         | `spec/{issue-id}-{name}`             | `spec/OG-009-sidenav`, `spec/009-sidenav` |
+| Feature Integration | `feat/{issue-id}-{name}`             | `feat/OG-009-sidenav`                     |
+| Milestone Branch    | `feat/{issue-id}-{name}/m{N}-{desc}` | `feat/OG-009-sidenav/m1-backend`          |
+| Hotfix              | `hotfix/{issue-id}-{desc}`           | `hotfix/OG-123-fix-login`                 |
+| Bugfix              | `fix/{issue-id}-{desc}`              | `fix/OG-456-null-check`                   |
+
+**Issue ID Formats** (Jira Integration):
+
+- Jira tickets: `OG-{###}` (e.g., `OG-009`, `OG-123`)
+- GitHub issues: `{###}` (e.g., `009`, `123`)
+- Internal tracking: `{###}` with optional prefix
+
+**Milestone Plan Structure** (in `plan.md`):
+
+```markdown
+| ID     | Branch         | Scope                    | Verification    | Depends On |
+| ------ | -------------- | ------------------------ | --------------- | ---------- |
+| M1     | m1-backend     | Entities, DAOs, Services | Unit tests pass | -          |
+| [P] M2 | m2-frontend    | React components         | Jest tests pass | -          |
+| M3     | m3-integration | API + E2E                | E2E tests pass  | M1, M2     |
+```
+
+**Workflow**:
+
+1. Create spec branch and specification PR (targets `develop`)
+2. Define Milestone Plan in `plan.md`
+3. For each milestone:
+   - Create milestone branch from feature branch or `develop`
+   - Implement tasks for that milestone
+   - Create PR with verification criteria
+   - Merge after review and tests pass
+4. Final integration PR (if using feature branch) merges to `develop`
```

Comment:
until main feature branch merged w/ develop, we branch off of feature. also, need to make sure we have a way to create specification clarification branches.